rules/rules-reviewed/camel3/camel2/xml-310-changes.windup.xml

```diff
@@ -0,0 +1,165 @@
+<?xml version="1.0"?>
+<ruleset xmlns="http://windup.jboss.org/schema/jboss-ruleset" id="xml-310-changes"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            Rules for changes between Camel 3.9 and Camel 3.10
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final" />
+        </dependencies>
+        <sourceTechnology id="camel" versionRange="[3.0,3.10)"/>
+        <targetTechnology id="camel" versionRange="[3.10,)" />
+    </metadata>
+    <rules>
+        <rule id="xml-removed-camel310-00001">
+            <when>
+                <project>
```

Comment:
This is a comment regarding all `project` conditions being used: there are two ways in windup to find dependencies:
* The `project` condition: this will look for dependencies in POM files.
* The `dependency` condition: this will look for artefacts within a JAR file (not every application must have a POM).

We usually apply both conditions when looking for dependencies, just in case the dependency is simply embedded in the JAR. To test this condition, we usually put some fake JARs in the corresponding test data folder. You can take a look at `hibernate-core-5.5.5.Final.jar` within `eap8/eap7/tests/data/hibernate-search-61`. The JAR simply contains a text file with a hash inside. In the `test.archive-metadata.txt` we then register those hashes for each dependency.

We do this so that the code doesn't get all bloated with actual dependencies.

Reply:
@jmle Can you point me to a good example in a windup.xml for how to do the `<or>` `<project>` `<dependency>` correctly to check for both conditions?   I'm looking throughout the rules and I don't see it used a lot - it's definitely not used in all cases of `<project>` conditions.
I think uber jars containing artifacts will be an edge case for us but I'm happy to add this - but this will demand a lot of changes and I want to make sure I get this correct.

Reply:
@cunningt you can take a look at an example [here](https://github.com/windup/windup-rulesets/blob/master/rules/rules-reviewed/quarkus/springboot/springboot-cloud-config-client-to-quarkus.windup.xml#L19). The test jar is [here](https://github.com/windup/windup-rulesets/blob/master/rules/rules-reviewed/quarkus/springboot/tests/data/springboot-cloud-config/spring-cloud-config-demo-0.0.1-SNAPSHOT.jar) and the file that maps it [here](https://github.com/windup/windup-rulesets/blob/master/rules/rules-reviewed/quarkus/springboot/tests/.windup/cache/nexus-indexer-data/test.archive-metadata.txt).

It is indeed quite a bit of work, I understand that. I don't know a lot about Camel, how can it be packaged? For instance, in a Spring Boot application, I guess one could find it embedded inside?

Reply:
Added dependencies, Added fake JARs to handle all dependencies, added test.archive-metadata.txt for both camel3 and camel4.    Ran tests, no issues seen.